### ISSUE TITLE
Support external lz4, zstd, blosc

### ISSUE THREAD

**Issue by haampie** [zarr-developers/numcodecs#464]
#### Problem description

This package seems to unconditionally build vendored copies of lz4, zstd and blosc. This (a) makes it hard to integrate this package with other libraries that pull in any of these libraries at a different version, and (b) update binary dependencies in case there are CVEs, and (c) deal with patches of those binary dependencies.

Can you please make this package work with an external/system lz4, zstd, and blosc? That'd be very helpful, thanks.

**Comment by martindurant** [comment]
I have mentioned elsehwere, but the package `cramjam` includes several useful and important compressors in one install, and ther tend to have good preformance compared to other versions. It does *not* support blosc, but that could be requested and would make numcodecs a much much simpler package to build and distribute.

**Comment by joshmoore** [comment]
Hi @haampie. I don't think anyone will disagree with your thesis here. It's more a matter of making it happen. Helping hands and suggestions like @martindurant's on getting us there are very welcome!

**Comment by BwL1289** [comment]
Linking #254. I know this is quite old. I'm experiencing this as well.

**Comment by martindurant** [comment]
Absolutely, thanks @milesgranger . The code here still references blosc1, so some compat work would be needed I think.

I really think that there is no need for numcodecs to be repeating the build process for these standard compressors and/or relying on system libraries. Some things will still need cython for custom algorithms, but the less the better.
